### DISCUSSION TITLE
Pyopencl on android

### DISCUSSION THREAD

**Discussion by kraken-d** [pyopencl/pyopencl#610] [Troubleshooting]
After lots of fixing I got to import pyopencl on my redmi note 7 pro with termux but now I am getting this error
$ ./opcl.py
/data/data/com.termux/files/usr/lib/python3.9/site-packages/pyopencl/__init__.py:265: CompilerWarning: Built kernel retrieved from cache. Original from-source build had warnings:
Build on <pyopencl.Device 'QUALCOMM Adreno(TM)' on 'QUALCOMM Snapdragon(TM)' at 0x74afc7bbb8> succeeded, but said:

Pass
  warn(text, CompilerWarning)
/data/data/com.termux/files/usr/lib/python3.9/site-packages/pyopencl/__init__.py:265: CompilerWarning: From-binary build succeeded, but resulted in non-empty logs:
Build on <pyopencl.Device 'QUALCOMM Adreno(TM)' on 'QUALCOMM Snapdragon(TM)' at 0x74afc7bbb8> succeeded, but said:

Error: Program not built!
  warn(text, CompilerWarning)
[0. 0. 0. ... 0. 0. 0.]
0.0
Plz help.

**Comment by inducer** [comment]
I'm not sure what to help with here. Assuming that `opcl.py` is derived from pyopencl's demo.py, this output (at least the zeros) is as expected. The rest are warning messages generated by the compiler that, it appears, are safe to ignore.

**Comment by Saikatsaha1996** [comment]
> After lots of fixing I got to import pyopencl on my redmi note 7 pro with termux but now I am getting this error $ ./opcl.py /data/data/com.termux/files/usr/lib/python3.9/site-packages/pyopencl/**init**.py:265: CompilerWarning: Built kernel retrieved from cache. Original from-source build had warnings: Build on <pyopencl.Device 'QUALCOMM Adreno(TM)' on 'QUALCOMM Snapdragon(TM)' at 0x74afc7bbb8> succeeded, but said:
> 
> Pass warn(text, CompilerWarning) /data/data/com.termux/files/usr/lib/python3.9/site-packages/pyopencl/**init**.py:265: CompilerWarning: From-binary build succeeded, but resulted in non-empty logs: Build on <pyopencl.Device 'QUALCOMM Adreno(TM)' on 'QUALCOMM Snapdragon(TM)' at 0x74afc7bbb8> succeeded, but said:
> 
> Error: Program not built! warn(text, CompilerWarning) [0. 0. 0. ... 0. 0. 0.] 0.0 Plz help.

Can you help please?

![IMG_20220916_104454](https://user-images.githubusercontent.com/72664192/190571088-61faa397-a197-480b-b6b0-cc7096cb83ca.jpg)
![IMG_20220916_115724](https://user-images.githubusercontent.com/72664192/190571257-415dd346-b450-48eb-ba08-51554613f865.jpg)

**Comment by inducer** [comment]
Your specific error message (that `clGetDeviceAndHostTimer` thing) indicates that you've got a mismatch between CL headers and the symbols provided by the ICD loader. You can make pyopencl ignore the CL version in the header using
```
./configure.py --cl-pretend-version=2.0
```
(or similar) before building. If you rebuild in the same source tree, make sure to blow away `build` before doing so.

**Reply by Saikatsaha1996** [reply]
Actually I downloaded it from `pip install pyopencl`

**Reply by inducer** [reply]
Do `git clone`, `cd pyopencl`, run the configure command, `pip install .`.

**Reply by Saikatsaha1996** [reply]
How do I solve this ## warning?
```
CompilerWarning: Non-empty compiler output encountered. Set the environment variable PYOPENCL_COMPILER_OUTPUT=1 to see more.
  warn("Non-empty compiler output encountered. Set the "
```
![IMG_20220917_072353](https://user-images.githubusercontent.com/72664192/190836330-f383ee7e-e6c2-416d-ae77-2d43abf77f5f.jpg)
